.github/workflows/ci.yml

```diff
@@ -0,0 +1,103 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  lint:
+    name: Static code analysis
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```

Comment:
With the large history of this repo, one more optimization could be shallow checkout.

Reply:
Excellent point. In fact we should update all CI/CD workflows, not only the new ones.